linode/manager
# upcoming: [DI-20935] - Support for tags filter in CloudPulse

## Description 📝

Added tags as an optional filter under service type - linode.

## Changes  🔄

List any change(s) relevant to the reviewer.

- Added tag selection component which is a multi-select.
- Added subtitle as 'optional' for tags.
- Made tag filter as an optional dependency for resources.
- Passed tag as an x-filter to filter resources.
- Handled preferences.
- Added unit test cases.
- Updated mocks for handling tags.


## Target release date 🗓️

14-Jan-2025


## Preview 📷

https://github.com/user-attachments/assets/860c8e5b-8f8e-4c70-944f-e3ac7c7c5561


## How to test 🧪

### Verification steps
- Navigate to monitor tab.
- Select a dashboard under service type - linode.
- You will be able to see tags filter listed as part of filters.
- Select any number of tags, select a region, you will see resources filtered out based on both the filters.
- Now, unselect all tags, choose region and resources, you will still be able to observe metrics as tags is an optional filter.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>

---
